src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminAreasOfLawService.java

```diff
@@ -53,6 +70,17 @@ public AreaOfLaw createAreaOfLaw(final AreaOfLaw areaOfLaw) {
         return new AreaOfLaw(areasOfLawRepository.save(createNewAreaOfLawEntityFromModel(areaOfLaw)));
     }
 
+    public void deleteAreaOfLaw(final Integer areaOfLawId) {
+        try {
+            ensureAreaOfLawIsNotInUse(areaOfLawId);
+            areasOfLawRepository.deleteById(areaOfLawId);
+        } catch (EmptyResultDataAccessException ex) {
```

Comment:
Do we need the two catch blocks here now that we manually throw a ListItemInUseException? 

If we are worried about a `find---` JPA method returning an exception (EmptyResultDataAccessException) we can wrap an optional around it, like: 

`List<CourtAreaOfLaw> getCourtAreaOfLawByAreaOfLawId(Integer areaOfLawId);` - for example 

Then on the method below, you can check if any rows are present for each by calling the .isPresent() method on it? Food for thought! 

Prob not a biggy, but I think it would remove these lines, and the only changes would be .isEmpty to .isPresent